staging/src/k8s.io/kubectl/pkg/cmd/config/unset.go

```diff
@@ -80,21 +88,44 @@ func (o unsetOptions) run(out io.Writer) error {
 		return err
 	}
 
-	steps, err := newNavigationSteps(o.propertyName)
-	if err != nil {
-		return err
-	}
-	err = modifyConfig(reflect.ValueOf(config), steps, "", true, true)
-	if err != nil {
-		return err
+	if o.jsonPath {
+		// Convert api config to apiv1 config, so we can use jsonpath properly
+		v1Config := &clientcmdapiv1.Config{}
+		if err := clientcmdapiv1.Convert_api_Config_To_v1_Config(config, v1Config, nil); err != nil {
+			return err
+		}
+
+		if err := modifyConfigJson(v1Config, o.propertyName, "", true, false, false); err != nil {
+			return err
+		}
+
+		// Convert the apiv1 config back to an api config to write back out
+		finalConfig := clientcmdapi.NewConfig()
+		if err := clientcmdapiv1.Convert_v1_Config_To_api_Config(v1Config, finalConfig, nil); err != nil {
+			return err
+		}
+		config = finalConfig
+	} else {
+		if _, err := fmt.Fprintln(o.streams.ErrOut, "Warning: usage of dot delimited path for setting config values is deprecated, please use jsonpath syntax instead."); err != nil {
```

Comment:
Does it make sense to deprecate the simple dot syntax people are used to? I would expect both to work without issue.